src/sentry/tasks/backfill_seer_grouping_records.py

```diff
@@ -159,64 +159,63 @@ def backfill_seer_grouping_records(
-            ),
-            Condition(Column("timestamp", entity=events_entity), Op.LT, time_now),
-        ],
-        orderby=[OrderBy(Column("group_id"), Direction.ASC)],
-    )
-
-    request = Request(
-        dataset=Dataset.Events.value,
-        app_id=Referrer.GROUPING_RECORDS_BACKFILL_REFERRER.value,
-        query=query,
-        tenant_ids={
-            "referrer": Referrer.GROUPING_RECORDS_BACKFILL_REFERRER.value,
-            "cross_org_query": 1,
-        },
-    )
+    # TODO(jangjodi): Only query per group if it has over 1 million events
+    snuba_event_data, group_id_batch_all = [], copy.deepcopy(group_id_batch)
+    for group_id in group_id_batch_all:
+        query = Query(
+            match=events_entity,
+            select=[
+                Column("group_id"),
+                Column("event_id"),
+            ],
+            where=[
+                Condition(Column("project_id"), Op.EQ, project.id),
+                Condition(Column("group_id"), Op.EQ, group_id),
+                Condition(
+                    Column("timestamp", entity=events_entity), Op.GTE, time_now - timedelta(days=90)
+                ),
+                Condition(Column("timestamp", entity=events_entity), Op.LT, time_now),
+            ],
+            limit=Limit(1),
+        )
 
-    with metrics.timer(f"{BACKFILL_NAME}.bulk_snuba_queries", sample_rate=1.0):
-        result = bulk_snuba_queries(
-            [request], referrer=Referrer.GROUPING_RECORDS_BACKFILL_REFERRER.value
+        request = Request(
+            dataset=Dataset.Events.value,
+            app_id=Referrer.GROUPING_RECORDS_BACKFILL_REFERRER.value,
+            query=query,
+            tenant_ids={
+                "referrer": Referrer.GROUPING_RECORDS_BACKFILL_REFERRER.value,
+                "cross_org_query": 1,
+            },
         )
 
-    if result and result[0].get("data"):
-        rows: list[GroupEventRow] = result[0]["data"]
-
-        # Log if any group does not have any events in snuba and skip it
-        if len(rows) != len(group_id_batch):
-            row_group_ids = {row["group_id"] for row in rows}
-            for group_id in group_id_batch:
-                if group_id not in row_group_ids:
-                    logger.info(
-                        "tasks.backfill_seer_grouping_records.no_snuba_event",
-                        extra={
-                            "organization_id": project.organization.id,
-                            "project_id": project_id,
-                            "group_id": group_id,
-                        },
-                    )
-                    group_id_batch.remove(group_id)
-                    del group_id_message_batch_filtered[group_id]
+        with metrics.timer(f"{BACKFILL_NAME}.bulk_snuba_queries", sample_rate=1.0):
+            result = bulk_snuba_queries(
+                [request], referrer=Referrer.GROUPING_RECORDS_BACKFILL_REFERRER.value
```

Comment:
You could potentially batch several of these together here and save yourself some time